File: Recursion/Pattern3/Pattern3.java
import java.util.*;
public class Pattern3 {
    public static void main(String[] args) {
        List<String> ret=fun("hell",0,0,new ArrayList<>());
        System.out.println(ret);
    }
    public static List<String> fun(String s,int start, int end, List<String> lst){
        if (start==s.length()){
            return lst;
        }
        if (end==s.length()){
            return fun(s,start+1,start+1,lst);
        }
        String s2=s.substring(start,end+1);
        lst.add(s2);
        fun(s,start,end+1,lst);
        return lst;
    }
}

File: Recursion/Patterns/Pattern3/Pattern3.py
def fun(start,end,string,lst):
    if start==len(string):
        return lst
    if end==len(string):
        return fun(start+1,start+1,string,lst)
    ss=string[start:end+1]
    lst.append(ss)
    fun(start,end+1,string,lst)
    return lst

out=fun(0,0,"hell",[])
print(out)